### PR TITLE
chore: Migrate sub components into Plugin Action Editor

### LINKED ISSUE #34324: [Task]: Modularise Plugin Action Editor
By combining the ApiEditor and QueryEditor components, we will create a new Plugin Action Editor component that would be composable and configurable to allow for use in all IDEs. 

https://www.notion.so/appsmith/Modularisation-Plugin-Action-Editor-4b306360bad34d2fb2d80e32c08bc4cb?pvs=4

- [x] #36152
- [x] #36153
- [x] #36154
- [x] #36155

The end result of this modularisation would allow for a composition of Plugin actions as so:

```
const AppPluginActionEditor {
	return (
		<PluginActionProvider {...}>
			<PluginActionEditor>
				<AppPluginActionToolbar />
				<PluginActionForm />
				<AppPluginActionBottomView />
			</PluginActionEditor>
		<ActionProvider>
	)
}


const WorkflowPluginActionEditor {
	return (
		<PluginActionProvider {...}>
			<PluginActionEditor>
				<WorkflowPluginActionToolbar />
				<PluginActionForm />
				<WorkflowPluginActionBottomView />
			</PluginActionEditor>
		<ActionProvider>
	)
}


const ModulePluginActionEditor {
	return (
		<PluginActionProvider {...}>
			<PluginActionEditor>
				<ModulePluginActionToolbar />
				<PluginActionForm />
				<PluginInputs />
				<ModulePluginActionBottomView />
			</PluginActionEditor>
		<ActionProvider>
	)
}
```

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

- Moving older components, utils and constants into the Plugin Action Editor to complete the modularisation project.
- Update Plugin Action Editor to expect an action ID instead of fetching it from the URL.
- Add a Readme

Fixes #34324 

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11363303457>
> Commit: c36241f8cb19d57365c2555a02284b72aa849a67
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11363303457&attempt=3" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 16 Oct 2024 12:40:23 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Introduced a new utility function `formatBytes` for byte formatting.
  - Added comprehensive documentation for the Plugin Action Editor module.

- **Improvements**
  - Simplified action retrieval logic in the Plugin Action Editor.
  - Enhanced import organization across various components for better maintainability.

- **Bug Fixes**
  - Resolved import path issues for several components and constants to ensure correct functionality.

- **Chores**
  - Updated multiple import paths to reflect the new directory structure for improved modularity.

### CODE REVIEW

File: app/client/src/PluginActionEditor/store/constants.ts
Comment (ankitakinger) [inline app/client/src/PluginActionEditor/store/constants.ts:1]: why are `constants` a part of `store` folder?

File: app/client/src/PluginActionEditor/PluginActionEditor.tsx [hunk 1]
```diff
@@ -20,15 +18,15 @@ import { useActionSettingsConfig } from "./hooks";
...
+  actionId: string;
 }
 
 const PluginActionEditor = (props: ChildrenProps) => {
-  const { pathname } = useLocation();
-
   const isEditorInitialized = useIsEditorInitialised();
 
-  const entity = identifyEntityFromPath(pathname);
-  const action = useSelector((state) => getActionByBaseId(state, entity.id));
```
Comment (ankitakinger) [inline app/client/src/PluginActionEditor/PluginActionEditor.tsx:31]: why did we need to change this?

File: app/client/src/PluginActionEditor/store/constants.ts
Comment (hetunandu) [reply app/client/src/PluginActionEditor/store/constants.ts:1]: The constant in there is only a store concern.

File: app/client/src/PluginActionEditor/PluginActionEditor.tsx (hunk 1 above)
Comment (hetunandu) [reply app/client/src/PluginActionEditor/PluginActionEditor.tsx:31]: It was a callout from @ashit-rath long back. We do not want to couple Action selection with the module. This way, it is possible to have a different way of selecting an action that is not always based on the url

File: app/client/src/PluginActionEditor/components/PluginActionForm/components/ApiEditor/APIEditorForm.tsx [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import React from "react";
-import CommonEditorForm from "./CommonEditorForm";
-import { usePluginActionContext } from "PluginActionEditor";
+import CommonEditorForm from "../CommonEditorForm";
+import { usePluginActionContext } from "PluginActionEditor/PluginActionContext";
```
Comment (ankitakinger) [inline app/client/src/PluginActionEditor/components/PluginActionForm/components/ApiEditor/APIEditorForm.tsx:3]: relative import?

File: app/client/src/PluginActionEditor/components/PluginActionForm/components/CommonEditorForm/components/EmbeddedDatasourcePathField.tsx [hunk 3]
```diff
@@ -42,15 +42,15 @@ import {
   getDatasource,
   getDatasourcesByPluginId,
 } from "ee/selectors/entitiesSelector";
-import { extractApiUrlPath } from "transformers/RestActionTransformer";
+import { extractApiUrlPath } from "PluginActionEditor/transformers/RestActionTransformer";
```
Comment (ankitakinger) [inline app/client/src/PluginActionEditor/components/PluginActionForm/components/CommonEditorForm/components/EmbeddedDatasourcePathField.tsx:45]: relative import?

File: app/client/src/PluginActionEditor/components/PluginActionForm/components/CommonEditorForm/components/EmbeddedDatasourcePathField.tsx [hunk 4]
```diff
@@ -42,15 +42,15 @@ import {
...
 import { getCurrentAppWorkspace } from "ee/selectors/selectedWorkspaceSelectors";
 import { Text } from "@appsmith/ads";
 import { TEMP_DATASOURCE_ID } from "constants/Datasource";
 import LazyCodeEditor from "components/editorComponents/LazyCodeEditor";
 import { getCodeMirrorNamespaceFromEditor } from "utils/getCodeMirrorNamespace";
 import { isDynamicValue } from "utils/DynamicBindingUtils";
 import { isEnvironmentValid } from "ee/utils/Environments";
-import { DEFAULT_DATASOURCE_NAME } from "constants/ApiEditorConstants/ApiEditorConstants";
+import { DEFAULT_DATASOURCE_NAME } from "PluginActionEditor/constants/ApiEditorConstants";
```
Comment (ankitakinger) [inline app/client/src/PluginActionEditor/components/PluginActionForm/components/CommonEditorForm/components/EmbeddedDatasourcePathField.tsx:53]: relative import?

File: app/client/src/ce/PluginActionEditor/components/PluginActionResponse/hooks/usePluginActionResponseTabs.tsx [hunk 5]
```diff
@@ -24,9 +24,9 @@ import {
   isActionRunning,
 } from "PluginActionEditor/store";
 import { doesPluginRequireDatasource } from "ee/entities/Engine/actionHelpers";
-import useShowSchema from "components/editorComponents/ActionRightPane/useShowSchema";
-import Schema from "components/editorComponents/Debugger/Schema";
-import QueryResponseTab from "pages/Editor/QueryEditor/QueryResponseTab";
+import useShowSchema from "PluginActionEditor/components/PluginActionResponse/hooks/useShowSchema";
+import Schema from "PluginActionEditor/components/PluginActionResponse/components/Schema";
+import QueryResponseTab from "PluginActionEditor/components/PluginActionResponse/components/QueryResponseTab";
```
Comment (ankitakinger) [inline app/client/src/ce/PluginActionEditor/components/PluginActionResponse/hooks/usePluginActionResponseTabs.tsx:29]: relative imports?

**Review by ankitakinger — APPROVED** [review]
LGTM

**Pull request merged**